rust/libnewsboat/src/fslock.rs

```diff
@@ -97,4 +97,11 @@ impl FsLock {
             false
         }
     }
+
+    pub fn try_lock_ffi(&mut self, new_lock_path: &str, pid: &mut i64) -> bool {
```

Comment:
I would very much prefer not to add wrappers to the libnewsboat crate: it muddles the API, and we'll have to track them down later when they aren't needed anymore. (The compiler won't help with that because `pub` methods are assumed to be used somewhere outside the crate.)

Instead, I'd write a function in the libnewsboat-ffi crate. It's uglier (one has to call `method(object, arg1, arg2)` instead of `object->method(arg1, args2)`), but I think it's better than polluting the API in libnewsboat crate. I think it's easier to *show* what I mean rather than describe it, so I pushed 2235c8fe into your branch. Feel free to squash it, drop it, or amend it to your liking — it's just a demonstration of what I envision, not a proposal for inclusion into this PR.

Reply:
> I would very much prefer not to add wrappers to the libnewsboat crate

Agreed. I had some trouble getting that to work. Thank you for the example!